synapse_data_warehouse/synapse_event/V2.51.0__node_event_table.sql

```diff
@@ -0,0 +1,92 @@
+USE SCHEMA {{ database_name }}.SYNAPSE_EVENT;
+
+CREATE OR REPLACE DYNAMIC TABLE node_event
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS
+        WITH latest_unique_rows AS (
+            SELECT
+                CHANGE_TYPE,
+                CHANGE_TIMESTAMP,
+                CHANGE_USER_ID,
+                SNAPSHOT_TIMESTAMP,
+                ID,
+                BENEFACTOR_ID,
+                PROJECT_ID,
+                PARENT_ID,
+                NODE_TYPE,
+                CREATED_ON,
+                CREATED_BY,
+                MODIFIED_ON,
+                MODIFIED_BY,
+                VERSION_NUMBER,
+                FILE_HANDLE_ID,
+                NAME,
+                IS_PUBLIC,
+                IS_CONTROLLED,
+                IS_RESTRICTED,
+                SNAPSHOT_DATE,
+                EFFECTIVE_ARS,
+                ANNOTATIONS,
+                DERIVED_ANNOTATIONS,
+                VERSION_COMMENT,
+                VERSION_LABEL,
+                ALIAS,
+                ACTIVITY_ID,
+                COLUMN_MODEL_IDS,
+                SCOPE_IDS,
+                ITEMS,
+                REFERENCE,
+                IS_SEARCH_ENABLED,
+                DEFINING_SQL,
+                INTERNAL_ANNOTATIONS,
+                VERSION_HISTORY,
+                PROJECT_STORAGE_USAGE
+            FROM
+                {{ database_name }}.synapse_raw.nodesnapshots --noqa: TMP
+            WHERE
+                snapshot_timestamp >= CURRENT_TIMESTAMP - INTERVAL '30 DAYS'
+            QUALIFY ROW_NUMBER() OVER (
+                    PARTITION BY id, version_number
```

Comment:
I know this is draft mode, but take a look at all the questions we would like to answer.

Specifically, entities can be updated without bumping the version - I think there is a question surrounding that

Reply:
We also want to capture DELETE events, right? Something I noticed is that there are no DELETE events with `version_number` as part of the entity metadata. 

<img width="958" alt="image" src="https://github.com/user-attachments/assets/0ee10092-e343-41ab-87d4-f873a6e73991" />

I looked throuhg some of these and they actually don't have much useful information, not even `node_type` 🤔 Might be something worth asking the platform team about too.

<img width="956" alt="image" src="https://github.com/user-attachments/assets/f2f5bcf7-15c2-4ac7-8e64-ad5e757e3765" />

Reply:
I would try and answer all the questions as listed in the Jira ticket with this table.

Reply:
@thomasyu888

> Specifically, entities can be updated without bumping the version - I think there is a question surrounding that

`version_number` uniquely enumerates updates to a node, even when a Synapse user is updating a previous version of the entity. See the Slack thread here: https://sagebionetworks.slack.com/archives/C05PBT2FUHF/p1747334513248759

Reply:
Addressed by [d9fc4b6](https://github.com/Sage-Bionetworks/snowflake/pull/205/commits/d9fc4b677cccbaf0c023693a20a451e8076aac94).